pkg/api/steve/proxy/path.go

```diff
@@ -0,0 +1,70 @@
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var errBadResource = errors.New("unidentified resource")
+var errBadGVR = errors.New("invalid GVR")
+
+// gvrFromPath tries to parse a URL path and extract GVR information from it.
+// Version must be present.
+// Group may be empty for core discovery client paths.
+// Resource may be empty for non-core discovery client paths.
+// This won't handle paths where "namespaces" designates a custom Namespace kind. So this function is not generally
+// applicable for any paths including any resources.
+func gvrFromPath(path string) (*schema.GroupVersionResource, error) {
+	parts := pathParts(path)
+	// Allow paths for core resource discovery, such as "/api/v1". And disallow non-core short paths, such as "/apis/management.cattle.io".
+	if len(parts) < 2 || (len(parts) < 3 && parts[0] != "api") {
```

Comment:
Can you explain the second case here a bit more? I'm not clear on if we should be failing to extract `/apis/management.cattle.io` - that definitely has a visible group

Reply:
This one represents a malformed discovery call, since it doesn't have the version.
While we want to extract GVR for `"/api/v1"`, whose group happens to be an empty string, but it represents a normal discovery call for core resources.